### ISSUE TITLE
[salt-cloud][azure] WindowsAzureError: Unknown error (Temporary Redirect) http code 307

### ISSUE THREAD

**Issue by andrejohansson** [saltstack/salt#14783]
It seems that Azure have started sending temporary redirects and this is not handled by salt-cloud and the azure python sdk it's using.

```
[DEBUG   ] Return code was 1
[ERROR   ] There was a query error: Unknown error (Temporary Redirect)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/cloud/cli.py", line 388, in run
    ret = mapper.run_map(dmap)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1647, in run_map
    iterable=parallel_data
  File "/usr/lib/python2.7/multiprocessing/pool.py", line 251, in map
    return self.map_async(func, iterable, chunksize).get()
  File "/usr/lib/python2.7/multiprocessing/pool.py", line 558, in get
    raise self._value
WindowsAzureError: Unknown error (Temporary Redirect)
```

The issue is reported [here](https://github.com/Azure/azure-sdk-for-python/issues/129) with a fix.

**Comment by techhat** [comment]
@andrejohansson, is this still happening with the latest Azure SDK?

**Comment by techhat** [comment]
No answer, and the issue seems to be reported as solved in azure-sdk-for-python#129. Please re-open if necessary.
